rfcs/20200422-concurrent-global-scratchpad-mode/README.md

```diff
@@ -0,0 +1,174 @@
+# Proposal for introducing concurrent and global scratchpad mode
+
+## Background
+
+Currently, some primitive implementations in oneDNN require a temporary storage
+for computations, a.k.a. scratchpad, and the library supports two scratchpad
+policies: global and concurrent.
+
+In most cases, the scratchpad has a concurrent mode, whereby each thread
+allocates a new scratchpad storage during the instantiation of a primitive that
+has been defined without `USE_GLOBAL_SCRATCHPAD` flag in the library. The
+following is an example of an implementation definition with concurrent
+scratchpad:
+
+~~~cpp
+
+DECLARE_COMMON_PD_T("ncsp_bnorm:any", ncsp_batch_normalization_fwd_t);
+
+~~~
+
+In case a primitive implementation does contain the `USE_GLOBAL_SCRATCHPAD`
```

Comment:
This seems like a violation of our cmake option. Basically, if the user specifies `DNNL_ENABLE_CONCURRENT_EXEC`, we still create primitives that are not thread safe? Do we know why this appeared?

Reply:
From what I understand, `DNNL_ENABLE_CONCURRENT_EXEC` disregards whatever the primitive implementation says in its definition (c.f. [scratchpad](https://github.com/oneapi-src/oneDNN/blob/master/src/common/scratchpad.cpp#L122) )

Reply:
my bad I had misread the code.